### PR TITLE
[kube-prometheus-stack] Align Grafana and Prometheus scrape intervals

### LINKED ISSUE #394: [kube-prometheus-stack] Scrape interval misconfigured in Grafana's default data source
**Describe the bug**

The [`grafana-datasource` ConfigMap](https://github.com/prometheus-community/helm-charts/blob/b643f58714a063e5a0ddf6832cd00ed29c0f4fab/charts/kube-prometheus-stack/templates/grafana/configmaps-datasources.yaml) doesn't define any scrape interval for Prometheus, which means the [default value of `15s`](https://grafana.com/docs/grafana/latest/datasources/prometheus/) is applied.

However, Prometheus' global scrape interval is `30s`:

```console
$ kubectl -n monitoring exec sts/prometheus-prometheus-stack-kube-prom-prometheus -c prometheus -- cat /etc/prometheus/config_out/prometheus.env.yaml
global:
  evaluation_interval: 30s
  scrape_interval: 30s
...
```

This makes the results of `rate()` functions used together with `[$__interval]` imprecise, or sometimes just plain wrong.

**Proposal**

I believe it would be sensible to default this value to `30s` if the user didn't explicitly define a scrape interval in the [chart's values](https://github.com/prometheus-community/helm-charts/blob/52dfc37b648001202edd9179effaec6aa188d887/charts/kube-prometheus-stack/values.yaml#L1659-L1670).

Here's how this can be achieved: https://github.com/grafana/grafana/issues/12656#issuecomment-435307645

**Other references**

See this blog post for a reference about why defining the scrape interval correctly is important: https://grafana.com/blog/2020/09/28/new-in-grafana-7.2-__rate_interval-for-prometheus-rate-queries-that-just-work/

_Helm and Kubernetes versions are irrelevant here._

### PR DESCRIPTION
**Pull request by antoineco**
#### What this PR does / why we need it:

Full description in the issue referenced below

*tl;dr:* it is important to ensure the scrape interval defined in the Grafana data source matches the one configured by the Prometheus operator. Right now, default values do not match ([`15s`](https://grafana.com/docs/grafana/latest/datasources/prometheus/) vs. [`30s`](https://github.com/prometheus-operator/prometheus-operator/blob/release-0.44/pkg/prometheus/promcfg.go#L180-L183)).

#### Which issue this PR fixes

Fixes #394

#### Special notes for your reviewer:

I'm bumping the _minor_ part of the chart's semver because the change does have an effect on the current default.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by antoineco** [comment]
Just to prove the change works as intended:

image

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 12.5.0
+version: 12.6.0
```
Comment (gkarthiks) [inline charts/kube-prometheus-stack/Chart.yaml:21]: lgtm, but would it be an increment to minor or patch?

Comment (antoineco) [reply charts/kube-prometheus-stack/Chart.yaml:21]: I wasn't sure 100% myself. As mentioned in the PR description, I though _minor_ was appropriate here, what do you think?

> I'm bumping the _minor_ part of the chart's semver because the change does have an effect on the current's default.

**Review by bismarck — APPROVED** [review]
LGTM. This was a good catch.
